### PR TITLE
Hacktoberfest - [JENKINS-62446]Test coverage for tool.maven and tool.mavenGlobalConfig

### PR DESCRIPTION
**Pull request by aHenryJard**
See [JENKINS-62446](https://issues.jenkins-ci.org/browse/JENKINS-62446).
(part of Hacktoberfest if possible :) )

This PR is the test coverage for issue https://github.com/jenkinsci/jenkins/pull/5008

I don't know how to deal with a fix that is not merge on Jenkins yet. Any help is welcome :)

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
Tests are a good thing, thanks! Ideally it should have been a demo with automated demo verification in CI (kudos to @v1v), but a test also does the job

**Comment by aHenryJard** [comment]
> Tests are a good thing, thanks! Ideally it should have been a demo with automated demo verification in CI (kudos to @v1v), but a test also does the job

I have no idea on what is the "automated demo verification in CI", do you have a link to something so I can see and learn please ?

**Comment by oleg-nenashev** [comment]
@aHenryJard please see https://github.com/jenkinsci/configuration-as-code-plugin/blob/b53eb77810a9f1a0b8871b21d98d25206f8e0d9f/demos/README.adoc#contributing for the guidelines and links

File: integrations/src/test/resources/io/jenkins/plugins/casc/GlobalToolGlobalMavenTest.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+tool:
+  maven:
+    installations:
+      - name: "custom"
+        properties:
+          - installSource:
+              installers:
+                - maven:
+                    id: "3.6.3"
+  # TODO when https://github.com/jenkinsci/jenkins/pull/5008 is merged and released
```
Comment (timja) [inline integrations/src/test/resources/io/jenkins/plugins/casc/GlobalToolGlobalMavenTest.yaml:10]: you can add an incrementals dependency in jenkins.version 

https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/cli/2.263-rc30473.618ccc39c1b0/

`2.263-rc30473.618ccc39c1b0`

**Comment by oleg-nenashev** [comment]
Will integrate the current pull request version. It is net-positive for the tests, and it will take a while to get the Jenkins core fix in the integration test anyway. So we can update the test in a follow-up. Thanks @aHenryJard !

**Pull request merged**